table/tables/cache_test.go

```diff
@@ -323,3 +323,44 @@ func TestCacheTablePointGet(t *testing.T) {
 	tk.MustQuery("select * from cache_point where u=13").Check(testkit.Rows("3 13 103"))
 	tk.MustQuery("select * from cache_point where id=2").Check(testkit.Rows("2 12 999"))
 }
+
+func TestCacheTableBatchPointGet(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create  table bp_cache_tmp1 (id int primary key auto_increment, u int unique, v int)")
+	tk.MustExec("insert into bp_cache_tmp1 values(1, 11, 101)")
+	tk.MustExec("insert into bp_cache_tmp1 values(2, 12, 102)")
+	tk.MustExec("insert into bp_cache_tmp1 values(3, 13, 103)")
+	tk.MustExec("insert into bp_cache_tmp1 values(4, 14, 104)")
+
+	// check point get out transaction
+	tk.MustQuery("select * from bp_cache_tmp1 where id in (1, 3)").Check(testkit.Rows("1 11 101", "3 13 103"))
+	tk.MustQuery("select * from bp_cache_tmp1 where u in (11, 13)").Check(testkit.Rows("1 11 101", "3 13 103"))
+	tk.MustQuery("select * from bp_cache_tmp1 where id in (1, 3, 5)").Check(testkit.Rows("1 11 101", "3 13 103"))
+	tk.MustQuery("select * from bp_cache_tmp1 where u in (11, 13, 15)").Check(testkit.Rows("1 11 101", "3 13 103"))
```

Comment:
This code branch does not actually cover the code you changed.

There are two cases the point get plan is generated.

```
select * from bp_cache_tmp1 whre u = 4
select * from bp_cache_tmp1 where u > 3 and u < 5
```

Although they're both point get, the code path is different.
When range query length is 1 and converted to point get, it's special case.